wsagent/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/impl/ProjectServiceApi.java

```diff
@@ -221,7 +221,8 @@ public ProjectConfigDto updateProject(String wsPath, ProjectConfigDto projectCon
       throws NotFoundException, ConflictException, ForbiddenException, ServerException, IOException,
           BadRequestException {
     if (wsPath != null) {
-      projectConfigDto.setPath(absolutize(wsPath));
+      wsPath = absolutize(wsPath);
```

Comment:
Again, what was the bug here?

Reply:
`absolutize` ensure that project path starts with `/`. 
Without that it leads it emitting `ProjectCreatedEvent` for already registered projects.

Have a look at code below:
```
    boolean registeredEarly = projectManager.isRegistered(wsPath);
    RegisteredProject updated = projectManager.update(projectConfigDto);
```
Without `absolutize` registration checking and updating will be performed for different project paths.